### ISSUE TITLE
How to use Subscriptions?

### ISSUE THREAD

**Issue by TSMMark** [rmosolgo/graphql-ruby#374]
I see this repo mentions the `subscription` root type many times, but I can not find any documentation or example on how to use it. Any pointers much appreciated. Thank you!

In case it's relevant, we are considering using Apollo on the front end.

**Comment by rmosolgo** [comment]
I found it a bit hard and don't have any suggestions. I'm still using our pre-existing (over-fetching based 😆 ) setup for live updates. 

But, I do have some previous effort in the area, it might serve as a starting point. This one served subscriptions over ActionCable:

https://github.com/rmosolgo/graphql-streaming#subscription

It boils down to a flow like this: 

- Root fields on `subscription` check a magical key on `context`. It holds an object where the field can register the subscription. 
- On the first execution of the query, that key holds a "real" subscriber object which registers the subscription
- On later executions (when re-evaluating the query to push an update), that key holds a "fake" subscriber object which implements the required API as no-ops 
- The subscriber object stores subscription state and provides an API to the rest of the application to trigger updates (I used ActionCable built-ins for this)

That's all I got! I think the general flow was ok, but subscription state management was hard.
